### ISSUE TITLE
Support new logging

### ISSUE THREAD

**Issue by jtcohen6** [dbt-labs/dbt-rpc#82]
`dbt-rpc` hooks into a lot of `logbook` magic today: [`logger.py`](https://github.com/dbt-labs/dbt-rpc/blob/main/dbt_rpc/rpc/logger.py)

I think there are two areas in particular that need spiking / investigation:
- Custom processors that enrich structured log events with server-specific metadata (`addr`, `http_method`, `request_id`, `method`, `response_code`, `context`)
- Every RPC request batches up `logs` specific to that request, and the list of `logs` is [included in the server response](https://github.com/dbt-labs/dbt-rpc/blob/70c1a6ba011be6a82d09b8af0b391184a288dd7e/dbt_rpc/rpc/task_handler.py#L406-L407). (I think that filtering happens [here](https://github.com/dbt-labs/dbt-rpc/blob/70c1a6ba011be6a82d09b8af0b391184a288dd7e/dbt_rpc/rpc/logger.py#L208-L210))

The desirable outcome may look like:
- Reimplementing core event-module functions, to enrich each event with additional fields (noted above)
- Using `EVENT_HISTORY` to batch up log events by `request_id`, and passing them into the response when ready / polled

### Feature parity

Once `dbt-rpc` supports new logging, we'll need to take stock of content/feature parity. I know a lot changed between legacy and new logging. There's one piece in particular that comes to mind:
- Legacy logging: `logbook` would enrich every row with the `unique_id` of the node it's relevant to, sticking it in `extra.unique_id` ([source](https://github.com/dbt-labs/dbt-core/blob/d1ae9dd37f5f403c5374eb079f4bd952f0793993/core/dbt/logger.py#L243-L249))
- New logging: all subclass events of [`NodeInfo`](https://github.com/dbt-labs/dbt-core/blob/d1ae9dd37f5f403c5374eb079f4bd952f0793993/core/dbt/events/base_types.py#L117-L121) include a much richer `node_info` dictionary, including `node_info.unique_id` — but this is certainly fewer events than the ones which previously had `extra.unique_id`. Most notably, it's missing `SQLQuery`, the actual SQL run for this model.

This is important for supporting the dbt Cloud IDE's ability to show logs per node run (e.g. [this screenshot](https://docs.getdbt.com/img/docs/dbt-cloud/50e939e-Screen_Shot_2019-11-05_at_9.08.38_PM.png)), which expects to filter on `extra.unique_id` today

**Comment by leahwicz** [comment]
The main goal of this is for the IDE to consume these logs so I think it would be 1) stdout 2) spec poll response. @jtcohen6 @davidharting correct me if I'm wrong here

**Comment by davidharting** [comment]
@leahwicz That sounds correct. The IDE still has access to the log files from dbt. Ultimately, we want "the same logs we get today" - which I believe is primarily stdout / stderror wrapped up into JSON and sent from the rpc server. 

I'm not precisely sure what "spec poll response" is.

**Comment by iknox-fa** [comment]
@davidharting Can you confirm that the rpc poll response is the desired location for the logs when you get a chance?

**Comment by davidharting** [comment]
@iknox-fa that is correct. We poll, and grab the logs from the poll response to render logs to the user. Once we get a "success" response from polling, we know we have the complete set of logs and continue to display that set of logs to the user any time that they refer back to that historical invocation

**Comment by iknox-fa** [comment]
Findings in brief: https://www.notion.so/dbtlabs/Structured-logging-with-dbt-rpc-79a87bef88394e39a97df5b0abe98af7
